### PR TITLE
Vb 4904 visit scheduler create controller and service layer

### PR DESCRIPTION
**Pull request by dhirajshettymoj**
What does this pull request do?
Adds controller and service layer for callbacks from Notify via notification-alerts service

What is the intent behind these changes?
VB-4904

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/controller/VisitNotifyController.kt [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.visitscheduler.config.ErrorResponse
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.notify.NotifyCallbackNotificationDto
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.notify.NotifyCreateNotificationDto
+import uk.gov.justice.digital.hmpps.visitscheduler.service.VisitNotifyService
+
+const val VISIT_NOTIFY_CONTROLLER_PATH: String = "/visits/notify"
```
[the remark this answers is not in the document]
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/controller/VisitNotifyController.kt:26]: not sure why I didn't do that. Will update

File: src/main/resources/db/migration/V3_34__create_visit_notify_history.sql [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    template_id            VARCHAR(36)      ,
+    template_version       VARCHAR(12)       ,
+    sent_at               timestamp,
+    completed_at          timestamp,
+    created_at            timestamp,
+
+    CONSTRAINT visit_notify_history_event_audit  FOREIGN KEY (event_audit_id) REFERENCES event_audit(id)
+);
+
+CREATE INDEX idx_notify_history_notification_id ON visit_notify_history(notification_id);
```
[the remark this answers is not in the document]
Comment (dhirajshettymoj) [reply src/main/resources/db/migration/V3_34__create_visit_notify_history.sql:17]: Added index on notification_type

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/audit/NotifyHistoryDto.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package uk.gov.justice.digital.hmpps.visitscheduler.dto.audit
+
+import io.swagger.v3.oas.annotations.media.Schema
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.enums.notify.NotifyNotificationType
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.enums.notify.NotifyStatus
+import uk.gov.justice.digital.hmpps.visitscheduler.model.entity.VisitNotifyHistory
+import java.time.LocalDateTime
+
+data class NotifyHistoryDto(
```
[the remark this answers is not in the document]
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/audit/NotifyHistoryDto.kt:9]: Added

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/controller/VisitNotifyController.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/controller/VisitNotifyController.kt:26]: Updated

**Review by adaviesMOJ — APPROVED** [review]

**Pull request merged**